tools/generic_includes.php

```diff
@@ -23,15 +23,21 @@
 $client     = new NDB_Client();
 $client->makeCommandLine();
 $client->initialize($configFile);
-$DB = Database::singleton();
+$DB     = Database::singleton();
+$config = NDB_Config::singleton();
 
-//allow instruments to find libraries
-require_once 'Utility.class.inc';
-
-// require all relevant OO class libraries
-require_once "Database.class.inc";
-require_once "NDB_Config.class.inc";
-require_once "NDB_BVL_Instrument.class.inc";
-require_once "Candidate.class.inc";
-
-?>
+/**
+ * Helper function that writes a string supplied in the parameters to a filepath
+ * specified in the parameters using the indicated mode. This function should be
+ * used as a standard in the tools PHP scripts.
+ *
+ * @param string $filePath Desired path for the generated file (including extension)
+ * @param string $output   String to be written to the file specified in $filePath
+ * @param string $mode     Creation mode of the file
+ */
+function writeToFile(string $filePath, string $output, string $mode = "w")
```

Comment:
I don't think this should be in the generic_includes file, it's specific to a couple scripts and if we ever want to modernize our code mixing function definitions with side-effects in the same file violates PSR-1 ("Basic Coding Standards").

"Files SHOULD either declare symbols (classes, functions, constants, etc.) or cause side-effects (e.g. generate output, change .ini settings, etc.) but SHOULD NOT do both."

Reply:
can I create a tools helper class then ?

Reply:
Sure.. (in fact, I think you need to because as far as I know you can't autoload functions in PHP.)

Reply:
it works though